pkg/manager/manager.go

```diff
@@ -64,6 +67,10 @@ type Manager interface {
 	// election was configured.
 	Elected() <-chan struct{}
 
+	// AddLogicalRunnableBuilder adds a controller builder to the manager, which is used to build
```

Comment:
So we make the usecase of "Controller is scoped to a cluster and runs once for each cluster" easy and ignore usecases where clusters interact with multiple clusters for now? If that is the case, why do we even bother to add the cluster into the reconcile.Request of the controller, it doesn't really need to know about that anymore?

Reply:
https://github.com/kubernetes-sigs/controller-runtime/pull/2207#discussion_r1118098878

Super fair to say that we want Controllers to be cluster-agnostic in the fullness of time, we just need to get there in a clean and safe way (somehow).

A similar concept of `AddLogicalRunnableBuilder`, but for watches maybe needs to be available in the manager, basically to make sure that we collect all the watches and handlers, and that we can close those when the cluster is closed.